packages/Angular/explorer-core/src/lib/generic-browser-list/generic-browser-list.component.ts

```diff
@@ -0,0 +1,181 @@
+      this.UpdateItemEvent.emit(updateItemEvent);
+    }
+    else{
+      this.sharedService.CreateSimpleNotification(`Unable to create folder ${resourceName}`, "error");
+    }
+
+  }
+
+  //todo - show a modal asking the user for a name to give the folder
+  public async createFolder(){
+    if(this.parentWillHandleAddFolder){
+      this.createFolderClickEvent.emit();
+      return;
+    }
+
+    let folderName: string = "sample new folder";
+    const md: Metadata = new Metadata();
+    const folderEntity: BaseEntity = await md.GetEntityObject<BaseEntity>(this.CategoryName);
+
+    folderEntity.NewRecord();
+    folderEntity.Set("Name", folderName);
+    folderEntity.Set("ParentID", this.selectedFolderID);
+    folderEntity.Set("Description", "Sample Description");
+
+    let saveResult: boolean = await folderEntity.Save();
+    if(saveResult){
+      this.sharedService.CreateSimpleNotification(`successfully created folder ${folderName}`, "info");
+
+      let folder: Folder = new Folder(folderEntity.Get("ID"), folderEntity.Get("Name"));
+      let item: Item<Folder> = new Item(folder, ItemType.Folder);
+      item.Description = folderEntity.Get("Description");
+      const updateItemEvent: UpdateItemEvent = new UpdateItemEvent(item, UpdateItemEventType.Add);
+      this.UpdateItemEvent.emit(updateItemEvent);
+    }
+    else{
+      this.sharedService.CreateSimpleNotification(`Unable to create folder ${folderName}`, "error");
+    }
+  }
+
+  public goHomeButtonClicked(){
+    this.backButtonClickEvent.emit();
+  }
+
+  private async deleteFolder(item: Item<any | Folder>){
+    if(item && item.Type === ItemType.Folder && this.parentWillHandleDeleteFolder){
+      this.deleteButtonClickEvent.emit(item);
+      return;
+    }
+    else if(item && item.Type === ItemType.Resource && !this.parentWillHandleDeleteResource){
+      this.deleteButtonClickEvent.emit(item);
+      return;
+    }
+
+    const folder: Folder = item.Data;
+
+    //the DB will throw an error if we attempt to delete a folder that has children
+    //i.e. sub folders or resources
+    //so the default behavior is to notify the user that the delete operation cannot
+    //go through 
+    const folderHasChildren: boolean = await this.doesFolderHaveChildren(folder.ID);
+    if(folderHasChildren){
+      this.sharedService.CreateSimpleNotification(`unable to delete folder ${folder.Name} because it has children`, "error");
+      return;
+    }
+
+    this.showLoader = true;
+    const md = new Metadata();
+    let folderEntity: BaseEntity = await md.GetEntityObject<BaseEntity>(this.CategoryName);
+    //@ts-ignore
+    let loadResult = await folderEntity.Load(folder.ID);
+    if(!loadResult){
+      this.sharedService.CreateSimpleNotification(`unable to fetch folder ${folder.Name}`, "error");
+      this.showLoader = false;
+      return;
+    }
+
+    let deleteResult = await folderEntity.Delete();
```

Comment:
@JS-BC  just seeing this now - per the above comment it does look like you've done this generically but perhaps the Dashboard browser is overriding this base functionality?

Reply:
If `parentWillHandleDeleteFolder`  or `parentWillHandleDeleteResource` is true, that tells the generic browser component that the parent component will handle deleting the given folder or whatever else, otherwise we fallback to using the logic inside the generic component to handle deletion 

There's similar logic  behind the variables `parentWillHandleAddFolder` and `parentWillHandleAddResource`